src/app/upgrade.cpp

```diff
@@ -305,6 +305,7 @@ namespace
             {u"BitTorrent/Session/UseRandomPort"_qs, u"Preferences/General/UseRandomPort"_qs},
             {u"BitTorrent/Session/uTPEnabled"_qs, u"Preferences/Bittorrent/uTP"_qs},
             {u"BitTorrent/Session/uTPRateLimited"_qs, u"Preferences/Bittorrent/uTP_rate_limited"_qs},
+            {u"BitTorrent/Session/DoNotDownload"_qs, u"Preferences/Downloads/DoNotDownload"_qs},
```

Comment:
Seems you've overthinked it... There is nothing to upgrade since you've just introduced this feature, isn't it?